async-nats/src/lib.rs

```diff
@@ -980,11 +980,29 @@ pub async fn connect_with_options<A: ToServerAddrs>(
         headers: true,
         no_responders: true,
     };
-    client
-        .sender
-        .send(Command::Connect(connect_info))
-        .await
-        .map_err(|_| io::Error::new(io::ErrorKind::Other, "failed to send connect"))?;
+    match options.auth {
+        Authorization::None => {}
+        Authorization::Token(token) => connect_info.auth_token = Some(token),
+        Authorization::UsernamePassword(user, pass) => {
+            connect_info.user = Some(user);
+            connect_info.pass = Some(pass);
+        }
+    }
+    connection_handler
+        .connection
+        .write_op(ClientOp::Connect(connect_info))
+        .await?;
+    connection_handler.connection.flush().await?;
+    match connection_handler.connection.read_op().await? {
+        Some(ServerOp::Error(err)) => {
+            return Err(io::Error::new(ErrorKind::InvalidInput, err.to_string()));
```

Comment:
What would the "other" case be here? At this stage there are no subscriptions etc so the only message we'll get are ping/pongs right?

```suggestion
```

Reply:
yes, but even manually handling `Pong`  only here looks worse than just saying "I don't care, let it be handled by the same logic and flow that handles it in all other cases". IMO this is the safest way, also being the shortest and most consistent.